lib/include/jxl/encode.h

```diff
@@ -630,6 +636,45 @@ JXL_EXPORT JxlEncoderStatus JxlEncoderAddImageFrame(
     const JxlEncoderFrameSettings* frame_settings,
     const JxlPixelFormat* pixel_format, const void* buffer, size_t size);
 
+/**
+ * TODO(firsching): add documentation
+ *
+ */
+typedef void (*JxlEncoderOutputCallback)(void* run_opaque, size_t pos,
+                                         size_t num_bytes);
+
+/**
+ * TODO(firsching): add documentation
+ *
+ */
+JXL_EXPORT JxlEncoderStatus
+JxlEncoderSetOutputCallback(JxlEncoderOutputCallback callback);
+
+/**
+ * TODO(firsching): add documentation
+ *
+ * @param frame_settings
+ * @return JXL_ENC_SUCCESS on success, JXL_ENC_ERROR on error
+ */
+JXL_EXPORT JxlEncoderStatus
+JxlEncoderChunkedImageFrameStart(const JxlEncoderFrameSettings* frame_settings);
+
+/**
+ * TODO(firsching): add documentation
+ *
+ * @param frame_settings
+ * @param pixel_format for pixels. Object owned by the caller and its contents
+ * are copied internally.
+ * @param x horizontal position of the top-left corner of the processed group.
+ * Must be divisible by 1024.
+ * @param y vertical postition of the top-left corner of the processed group.
+ * Must be divisible by 1024.
+ * @return JXL_EXPORT
+ */
+JXL_EXPORT JxlEncoderStatus JxlEncoderChunkedImageFrameAddPart(
```

Comment:
Why offsets divisible by 1024? Shouldn't that be either 256 (or rather kGroupDim) or 2048, depending on the granularity we want for this?
Also: how does the actual input buffer and its dimensions get passed?

Reply:
yes, 2048
Added the input buffer argument...